lib/inv_cg_quda.cpp

```diff
@@ -242,6 +244,14 @@ namespace quda {
 
     // whether to select alternative reliable updates
     bool alternative_reliable = param.use_alternative_reliable;
+    /**
+      When CG is used as a preconditioner, we do not want enable the `advanced features`, which includes:
+      - Reliable updates
+      - Pipelining
```

Comment:
Why wold we not want to use pipelining when CG is used a preconditioner?

Reply:
I should change the wording here. These features are turned off if CG is used as a preconditioner and `precondition_no_advanced_feature` is set to true. In `invert_quda.h`,
```
precondition_no_advanced_feature(param.schwarz_type == QUDA_ADDITIVE_SCHWARZ || param.schwarz_type == QUDA_ADDITIVE_MADWF_SCHWARZ)
```

Reply:
https://github.com/lattice/quda/pull/1061/commits/7e090a3c13613aee3586545a9289f60982c00cbd